File: src/main/java/com/brasilburger/model/Paiement.java
package com.brasilburger.model;

import com.brasilburger.model.enums.ModePaiement;
import java.time.LocalDateTime;

/**
 * Classe représentant un Paiement
 */
public class Paiement {
    private int id;
    private int commandeId;
    private double montant;
    private ModePaiement modePaiement;
    private LocalDateTime datePaiement;
    private boolean valide;
    private String referenceTransaction;

    public Paiement() {
        this.datePaiement = LocalDateTime.now();
        this.valide = false;
    }

    public Paiement(int commandeId, double montant, ModePaiement modePaiement) {
        this.commandeId = commandeId;
        this.montant = montant;
        this.modePaiement = modePaiement;
        this.datePaiement = LocalDateTime.now();
        this.valide = false;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getCommandeId() { return commandeId; }
    public void setCommandeId(int commandeId) { this.commandeId = commandeId; }

    public double getMontant() { return montant; }
    public void setMontant(double montant) { this.montant = montant; }

    public ModePaiement getModePaiement() { return modePaiement; }
    public void setModePaiement(ModePaiement modePaiement) { this.modePaiement = modePaiement; }

    public LocalDateTime getDatePaiement() { return datePaiement; }
    public void setDatePaiement(LocalDateTime datePaiement) { this.datePaiement = datePaiement; }

    public boolean isValide() { return valide; }
    public void setValide(boolean valide) { this.valide = valide; }

    public String getReferenceTransaction() { return referenceTransaction; }
    public void setReferenceTransaction(String referenceTransaction) { this.referenceTransaction = referenceTransaction; }

    @Override
    public String toString() {
        return "Paiement{" +
                "id=" + id +
                ", commandeId=" + commandeId +
                ", montant=" + montant +
                ", modePaiement=" + modePaiement +
                ", datePaiement=" + datePaiement +
                ", valide=" + valide +
                '}';
    }
}

File: src/Entity/Paiement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'paiements')]
class Paiement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\OneToOne(inversedBy: 'paiement', targetEntity: Commande::class)]
    #[ORM\JoinColumn(name: 'commande_id', referencedColumnName: 'id', nullable: false)]
    private ?Commande $commande = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private string $montant;

    #[ORM\Column(name: 'mode_paiement', length: 50)]
    private string $modePaiement;

    #[ORM\Column(name: 'date_paiement', type: 'datetime_immutable', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeImmutable $datePaiement;

    #[ORM\Column(type: 'boolean', options: ['default' => false])]
    private bool $valide = false;

    #[ORM\Column(name: 'reference_transaction', length: 255, nullable: true)]
    private ?string $referenceTransaction = null;

    public function __construct()
    {
        $this->datePaiement = new \DateTimeImmutable();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): self
    {
        $this->commande = $commande;

        return $this;
    }

    public function getMontant(): string
    {
        return $this->montant;
    }

    public function setMontant(string $montant): self
    {
        $this->montant = $montant;

        return $this;
    }

    public function getModePaiement(): string
    {
        return $this->modePaiement;
    }

    public function setModePaiement(string $modePaiement): self
    {
        $this->modePaiement = $modePaiement;

        return $this;
    }

    public function getDatePaiement(): \DateTimeImmutable
    {
        return $this->datePaiement;
    }

    public function setDatePaiement(\DateTimeImmutable $datePaiement): self
    {
        $this->datePaiement = $datePaiement;

        return $this;
    }

    public function isValide(): bool
    {
        return $this->valide;
    }

    public function setValide(bool $valide): self
    {
        $this->valide = $valide;

        return $this;
    }

    public function getReferenceTransaction(): ?string
    {
        return $this->referenceTransaction;
    }

    public function setReferenceTransaction(?string $referenceTransaction): self
    {
        $this->referenceTransaction = $referenceTransaction;

        return $this;
    }
}
